agent-inject/agent/annotations_test.go

```diff
@@ -628,6 +628,64 @@ func TestSecretCommandAnnotations(t *testing.T) {
 	}
 }
 
+func TestSecretErrorOnMissingKeyAnnotations(t *testing.T) {
+	tests := []struct {
+		annotations map[string]string
+		expectedKey string
+		expected    bool
+	}{
+		{
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foobar":  "test1",
+				"vault.hashicorp.com/error-on-missing-key-foobar": "True",
+			}, "foobar", true,
+		},
+		{
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foobar":  "test2",
+				"vault.hashicorp.com/error-on-missing-key-foobar": "false",
+			}, "foobar", false,
+		},
+		{
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foobar":  "test3",
+				"vault.hashicorp.com/error-on-missing-key-foobar": "false",
+			}, "foobar", false,
+		},
+		{
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foobar": "test4",
+			}, "foobar", false,
+		},
```

Comment:
It would be good to have a test case where `error-on-missing-key-` is set to something besides true or false.

Reply:
I've added a test to check a bad annotation value. But it only checks that an error occurs because `parseutil.ParseBool` doesn't wrap an error from `strconv`: https://github.com/mitchellh/mapstructure/blob/main/mapstructure.go#L741.